api/client.go

```diff
@@ -130,27 +129,16 @@ func (c *httpClient) Do(ctx context.Context, req *http.Request) (*http.Response,
 	}()
 
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	var body []byte
-	done := make(chan struct{})
 	go func() {
-		var buf bytes.Buffer
-		_, err = buf.ReadFrom(resp.Body)
-		body = buf.Bytes()
-		close(done)
-	}()
-
-	select {
-	case <-ctx.Done():
-		<-done
+		<-ctx.Done()
 		err = resp.Body.Close()
 		if err == nil {
 			err = ctx.Err()
 		}
-	case <-done:
-	}
+	}()
```

Comment:
Nothing is now waiting for the goroutine to finish, so assigning the error is pointless.
(This may not be important, but stylistically the code should not do pointless things)

Reply:
1. Do we need to do `resp.Body.Close()` now, as we are no longer creating a buffer?
2. The error is being returned in the end, should we still not assign it?